src/descriptor/mod.rs

```diff
@@ -68,6 +75,258 @@ pub enum Descriptor<Pk: MiniscriptKey> {
     ShWsh(Miniscript<Pk, Segwitv0>),
 }
 
+#[derive(Debug, Eq, PartialEq, Clone, Ord, PartialOrd, Hash)]
+pub enum DescriptorPublicKey {
+    ShortPub(DescriptorShortPub),
+    XPub(DescriptorXPub),
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Ord, PartialOrd, Hash)]
+pub struct DescriptorShortPub {
+    origin: Option<(bip32::Fingerprint, bip32::DerivationPath)>,
+    key: bitcoin::PublicKey,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Ord, PartialOrd, Hash)]
+pub struct DescriptorXPub {
+    origin: Option<(bip32::Fingerprint, bip32::DerivationPath)>,
+    xpub: bip32::ExtendedPubKey,
+    derivation_path: bip32::DerivationPath,
+    is_wildcard: bool,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct DescriptorKeyParseError(&'static str);
```

Comment:
I used this as a placeholder in my WIP PR, but we should probably give more contextualized and structural errors.

Reply:
Yeah i wanted to include it into our global `Error` enum at first but figured it'd be OK (we have per-file error eg for the policy compiler too).. On the other hand i think it's the only non-enum one. Will give it another look.

Reply:
Ok so i tried implementing that as a proper `DescriptorParseError` enum. And i actually end up with less contextualized errors and useless enum variants.

We might want at some point to have it as a variant of a generalistic DescriptorError placed in mod.rs (there is already the InterpreterError in satisfied_constraints.rs). Don't want to drastically increase this PR's diff though.

Reply:
Agreed. This can be done later